zingolib/src/commands/utils.rs

```diff
@@ -24,9 +28,19 @@ pub(super) fn parse_send_args(
 
                 let address = j["address"]
                     .as_str()
-                    .ok_or(CommandError::UnexpectedType)?
+                    .ok_or(CommandError::UnexpectedType(
+                        "address not a Str!".to_string(),
+                    ))?
                     .to_string();
-                let amount = j["amount"].as_u64().ok_or(CommandError::UnexpectedType)?;
+                let amount = if !j["amount"].is_number() {
+                    return Err(CommandError::NonJsonNumberForAmount(
+                        "amount is not a json::number::Number".to_string(),
+                    ));
+                } else {
+                    j["amount"].as_u64().ok_or(CommandError::UnexpectedType(
+                        "amount not a u64!".to_string(),
```

Comment:
No.   If `is_number` is false then any invocation of `as_u64` would be `None`.

The (small) benefit of checking the "numberness" first is that a user will get a different message for `OscarPepper` (which isn't a JSON Number) and `-64` which *is* a JSON Number but not representable as a `u64`.   It's not more safe or correct, since both `-64` and `OscarPepper` are already rejected, but it's (sligthly) more informative (about *why*).

Is it worth the complexity, and readability costs?   I don't know.  I think that fewer instances of `UnexpectedType` being presented to the consumer is pretty valuable.   Clearly this isn't a complete solution.